src/modules/Rear.rb

```diff
@@ -72,57 +73,75 @@ def GetUsbPartitions
+    #   'eval "${CONFIG_VAR[@]}"' and for wildcard names that should not be expanded by the shell
+    #   directly.
 
     # Convert List from Config File to YCP List
     def RearListToYCPList(rearlist)
-      ycplist = []
-      ycplisttmp = []
-
       return [] if rearlist == nil
 
-      # remove brakets
+      # remove parentheses
       rearlist = Builtins.regexpsub(Convert.to_string(rearlist), "^ *\\((.*)\\) *$", "\\1")
 
-      # split string seperated by spaces into a string list, respect backslash escaped blanks
-      ycplisttmp = Builtins.splitstring(rearlist, " ")
+      # split string separated by spaces into a string list, respect backslash escaped blanks
+      ycplisttmp = rearlist.split(" ")
+      ycplist = []
       buffer = ""
-      Builtins.foreach(ycplisttmp) do |elem|
-        length = Builtins.size(elem)
-        if Builtins.substring(elem, Ops.subtract(length, 1), 1) == "\\"
-          buffer = Ops.add(
-            Ops.add(
-              buffer,
-              Builtins.substring(elem, 0, Ops.subtract(length, 1))
-            ),
-            " "
-          )
+      ycplisttmp.each do |elem|
+        if elem[-1] == "\\"
+          buffer += elem[0...-1] + " "
         else
-          buffer = Ops.add(buffer, elem)
-          ycplist = Builtins.add(ycplist, buffer)
+          buffer += elem
+          ycplist.push(buffer)
           buffer = ""
         end
       end
 
       # remove empty elements
-      ycplist = Builtins.filter(ycplist) { |element| element != "" }
+      ycplist = ycplist.select { |elem| elem != "" }
 
       deep_copy(ycplist)
     end
 
-
     # Convert YCP List to Format for Config File
     def YCPListToRearList(ycplist)
-      ycplist = deep_copy(ycplist)
-      escaped = []
       # escape blanks in directories with a backslash
-      Builtins.foreach(ycplist) do |elem|
-        escaped = Builtins.add(
-          escaped,
-          Builtins.mergestring(Builtins.splitstring(elem, " "), "\\ ")
-        )
-      end
-      Ops.add(Ops.add("(", Builtins.mergestring(escaped, " ")), ")")
+      escaped = ycplist.map { |elem| elem.gsub(" ", "\\ ") }
+
+      "(" + escaped.join(" ") + ")"
+    end
+
+    # Convert Quoted List from Config File to YCP List
+    def RearQuotedListToYCPList(rearlist)
+      return [] if rearlist == nil
+
+      # remove parentheses
+      rearlist = Builtins.regexpsub(Convert.to_string(rearlist), "^ *\\((.*)\\) *$", "\\1")
+
+      # split string separated by spaces into a string list, respect shell quoting
+      ycplist = Shellwords.split(rearlist);
+
+      # remove empty elements
+      ycplist = ycplist.select { |elem| elem != "" }
+
+      deep_copy(ycplist)
+    end
+
+    # Convert YCP List to Quoted Format for Config File
+    def YCPListToRearQuotedList(ycplist)
+      # escape all commands by putting them in single quotes, instead of using Shellwords.escape
+      # which would escape each individual character thus making the result harder to read
```

Comment:
we already have method for such escaping so it is preffered to use it: https://github.com/yast/yast-yast2/blob/1629c3a520ebd8e8b93f1161f06d9e140d65b9ec/library/types/src/modules/String.rb#L57

Reply:
Updated to use `String.Quote()`.